tidy3d/plugins/autograd/README.md

```diff
@@ -97,61 +106,116 @@ import tidy3d as td
+| polyslab (including those with dilation or slanted sidewalls)     | `PolySlab.vertices`                                     |
+| regular mediums                                                   | `Medium.permittivity`, `Medium.conductivity`            |
+| spatially varying mediums (for topology optimization mainly)      | `CustomMedium.permittivity`, `CustomMedium.eps_dataset` |
+| groups of geometries with the same medium (for faster processing) | `GeometryGroup.geometries`                              |
+| complex and self-intersecting polyslabs                           | `ComplexPolySlab.vertices`                              |
+| dispersive materials                                              | `PoleResidue.eps_inf`, `PoleResidue.poles`              |
+| spatially dependent dispersive materials                          | `CustomPoleResidue.eps_inf`, `CustomPoleResidue.poles`  |
+
+The following components are traceable as outputs of the `td.SimulationData`
+
+| Data Type         | Traceable Attributes & Methods                                |
+| ----------------- | ------------------------------------------------------------- |
+| `ModeData`        | `amps`                                                        |
+| `DiffractionData` | `amps`                                                        |
+| `FieldData`       | `field_components`, `flux`                                    |
+| `SimulationData`  | `get_intensity(field_monitor)`, `get_poynting(field_monitor)` |
+
+We also support the following high-level features:
+
+- Compute gradients for objective functions that rely on multi-frequency data using a single broadband adjoint source.
+- Enable server-side gradient processing by setting `local_gradient=False` in the web functions.
+  This can significantly reduce data storage time.
+  However, exercise caution when using this feature with multi-frequency monitors and large design regions, as it may result in substantial data storage on our servers.
 
 We currently have the following restrictions:
 
-- Only 500 max structures containing tracers can be added to the `Simulation` to cut down on processing time. To bypass this restriction, use `GeometryGroup` to group structures with the same medium.
-- `web.run_async` for simulations with tracers does not return a `BatchData` but rather a `dict` mapping task name to `SimulationData`. There may be high memory usage with many simulations or a lot of data for each.
+- Only 500 max structures containing tracers can be added to the `Simulation` to cut down on processing time.
+  To bypass this restriction, use `GeometryGroup` to group structures with the same medium.
+- `web.run_async` for simulations with tracers does not return a `BatchData` but rather a `dict` mapping task name to `SimulationData`.
+  There may be high memory usage with many simulations or a lot of data for each.
 
```

Comment:
should we add broadband? Basically:

Tidy3D can handle objective functions over a single simulation under any of the following conditions for the monitors that the objective function output depends on:
1. Several monitors, all withthe same frequency.
2. One monitor with many frequencies where the data is being extracted out of a single coordinate (eg. single mode_index or direction will work, multiple will not).

If your optimization does not fall into one of these categories, you must split it into separate simulations and run them with `web.run_asnyc`. In all cases, the adjoint simulation bandwidth will be the same as the forward simulation. These limitations allow us to avoid the need to use methods that combine all adjoint sources into one simulation, which have the potential to degrade accuracy and increase the run time and cost significantly. That being said, we plan to offer support for more flexible and general broadband adjoint in the future.